store/store.go

```diff
@@ -1072,7 +1072,8 @@ func (s *Store) Download(name string, targetFn string, downloadInfo *snap.Downlo
 	if err := os.MkdirAll(filepath.Dir(targetFn), 0755); err != nil {
 		return err
 	}
-	w, err := os.Create(targetFn + ".partial")
+	resume := int64(0)
+	w, err := os.OpenFile(targetFn+".partial", os.O_APPEND|os.O_RDWR|os.O_CREATE, 0644)
```

Comment:
Thanks @zyga, this is indeed a bit dangerous, I changed it to use seek-after-open now.